src/docfx/template/TemplateEngine.cs

```diff
@@ -28,19 +28,46 @@ internal class TemplateEngine
         private readonly SearchIndexBuilder? _searchIndexBuilder;
         private readonly BookmarkValidator? _bookmarkValidator;
 
-        public TemplateEngine(
+        public static TemplateEngine CreateTemplateEngine(
+            ErrorBuilder errors,
+            Config? config,
+            PackageResolver? packageResolver,
+            string locale,
+            Package? package,
+            BookmarkValidator? bookmarkValidator = null,
+            SearchIndexBuilder? searchIndexBuilder = null,
+            bool fullBuild = true)
+        {
+            if (fullBuild)
```

Comment:
Consider use two static factory methods over a bool switch. With 2 static factory methods, we could remove these null checks.

Reply:
Done.